src/membership/ring.rs

```diff
@@ -0,0 +1,120 @@
+use std::{cmp::Ordering, collections::BTreeSet, hash::Hasher, ops::Bound};
+use twox_hash::XxHash64;
+
+#[derive(Debug, Clone)]
+pub struct Ring<T> {
+    seed: u64,
+    set: BTreeSet<SeededKey<T>>,
+}
+
+#[derive(Debug, Clone)]
+struct SeededKey<T> {
+    key: T,
+    seed: u64,
+}
+
+impl<T: AsRef<[u8]>> Ring<T> {
+    pub fn new(seed: u64) -> Self {
+        Self {
+            seed,
+            set: BTreeSet::new(),
+        }
+    }
+
+    pub fn contains(&self, key: T) -> bool {
+        // TODO(lucio): Remove this clone since we only need a `AsRef<[u8]>` to
```

Comment:
This TODO is done?

Reply:
no gonna have to leave it in couldn't find a way to get the ref to work :(